### PR TITLE
Ty 1932 part 5 xay net like fork join training

### PR DESCRIPTION
**Pull request by rustonaut**
Emulate the split-training nature of training with XaynNet.

### CODE REVIEW

File: dev-tool/src/list_net/train.rs [hunk 1]
```diff
@@ -59,16 +72,63 @@ pub struct TrainCmd {
...
 
     /// Dumps the initial parameters before any training was done.
     #[structopt(long)]
     dump_initial_parameters: bool,
+
+    /// Enables XaynNet Emulation mode.
+    ///
+    /// In this mode the list net is distributed to N users which
+    /// each train it using the given parameters and then the
+    /// separately trained list nets are merged. This is then
```
Comment (Robert-Steiner) [inline dev-tool/src/list_net/train.rs:89]:
```suggestion
    /// separately trained ListNets are then merged. This is then
```

File: dev-tool/src/list_net/cli_callbacks.rs [hunk 2]
```diff
@@ -9,16 +14,15 @@ use rayon::prelude::*;
...
+    /// The output used to store files (e.g. ListNet parameter).
+    pub(crate) file_output: FileOutput,
 
-    /// If true dumps the initial parameters.
-    ///
-    /// This can be useful if the initial parameters
-    /// have been freshly created using some random
-    /// initializer.
-    pub(crate) dump_initial_parameters: bool,
+    /// Implementation used to hint training progress to the user.
```
Comment (Robert-Steiner) [inline dev-tool/src/list_net/cli_callbacks.rs:24]:
```suggestion
    /// Implementation used to hint the training progress to the user.
```

File: dev-tool/src/list_net/cli_callbacks.rs [hunk 3]
```diff
@@ -195,39 +156,167 @@ impl TrainingController for CliTrainingController {
...
+    fn hint_train_progress(&self, _inc_process: u64) {}
+
+    fn hint_epoch_progress(&self, _inc_process: u64, _loss: f32) {}
+
+    fn hint_eval_progress(&self, _inc_process: u64, _mean_eval_cost: Option<f32>) {}
+
+    fn println(&self, _line: &str) {}
+}
+
+/// Training Feedback
```
Comment (Robert-Steiner) [inline dev-tool/src/list_net/cli_callbacks.rs:261]:
```suggestion
/// Training feedback
```

File: dev-tool/src/list_net/train.rs [hunk 4]
```diff
@@ -59,16 +72,63 @@ pub struct TrainCmd {
...
     #[structopt(long)]
     dump_every: Option<usize>,
 
     /// Dumps the initial parameters before any training was done.
     #[structopt(long)]
     dump_initial_parameters: bool,
+
+    /// Enables XaynNet Emulation mode.
+    ///
+    /// In this mode the list net is distributed to N users which
```
[the remark this answers is not in the document]
Comment (rustonaut) [reply dev-tool/src/list_net/train.rs:87]: Are you sure? Also is it appropriate to place a comma after users?

[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply dev-tool/src/list_net/train.rs]: Ah ok I didn't know that `it` means the users. 

> /// In this mode the ListNet is distributed to N users. Each user
/// trains it using the given parameters. Then the
/// separately trained ListNets are merged.

👍

[the remark this answers is not in the document]
Comment (rustonaut) [reply dev-tool/src/list_net/train.rs]: It wasn't too clear.

I now did apply all change requests. (Speculating that the proposed text will be ok I previously already pushed a commit containing it. :wink: )

**Review by Robert-Steiner — APPROVED** [review]
🚀

**Pull request merged**